detox/src/client/Client.test.js

```diff
@@ -279,14 +284,41 @@ describe('Client', () => {
     expect(client.ws.send).toHaveBeenCalledTimes(2);
   });
 
-  it(`execute() - "testFailed" result should throw`, async () => {
+  it(`execute() - "testFailed" result should throw with view hierarchy`, async () => {
+    await connect();
+    client.ws.send.mockReturnValueOnce(response("testFailed",  {details: "this is an error", viewHierarchy: 'mock-hierarchy'}, 1));
+    const call = invoke.call(invoke.IOS.Class('GREYMatchers'), 'matcherForAccessibilityLabel:', 'test');
+    try {
+      await client.execute(call);
+    } catch (ex) {
+      expect(ex).toBeDefined();
+      expect(ex.toString()).toContain('View Hierarchy:\nmock-hierarchy');
+    }
+  });
+
+  it(`execute() - "testFailed" result should throw with view-hierarchy hint`, async () => {
+    log.level.mockReturnValue(bunyan.INFO);
+
+    await connect();
+    client.ws.send.mockReturnValueOnce(response("testFailed",  {details: "this is an error", viewHierarchy: 'mock-hierarchy'}, 1));
+    const call = invoke.call(invoke.IOS.Class('GREYMatchers'), 'matcherForAccessibilityLabel:', 'test');
+    try {
+      await client.execute(call);
+    } catch (ex) {
+      expect(ex).toBeDefined();
+      expect(ex.toString()).toContain('use log-level verbose or higher');
+    }
+  });
+
+  it(`execute() - "testFailed" result should throw without a view hierarchy`, async () => {
     await connect();
-    client.ws.send.mockReturnValueOnce(response("testFailed",  {details: "this is an error"}, 1));
+    client.ws.send.mockReturnValueOnce(response("testFailed",  {details: "this is an error", viewHierarchy: undefined}, 1));
     const call = invoke.call(invoke.IOS.Class('GREYMatchers'), 'matcherForAccessibilityLabel:', 'test');
     try {
       await client.execute(call);
     } catch (ex) {
       expect(ex).toBeDefined();
```

Comment:
@noomorph note that this assertion was never enough. The test passes but the actual error was coming from the comparison of `log.level() <= bunyan.DEBUG` with `log.level()` returning `undefined`, and NOT the actual error being expected here. The additional mocks (see above) and tests are hence needed here, and thus fix this situation.